app/Services/BorrowBookService.php

```diff
@@ -0,0 +1,108 @@
+
+        $date_time= mktime(0, 0, 0, date("m"), date("d")+10, date("Y"));
+    
+        $book_borrow['book_id'] = $book_id;
+        $book_borrow['user_id'] = auth()->id();
+        $book_borrow['quantity'] = 1;
+        $book_borrow['status'] = 0;
+        $book_borrow['datetime_borrow'] = Carbon::now();
+        $book_borrow['datetime_return'] = date("Y/m/d H:i:s", $date_time);
+        $book_borrow['sub_total'] = $price *  $book_borrow['quantity'];
+
+        $add_borrow = BorrowedBook::create($book_borrow);
+    
+        if($add_borrow){
+            return redirect(route('book_borrow.index'));
+        }else{
+            return redirect()->back();
+        }
+    }
+
+    public function qtyPlus($borrowed_id,$quantity,$book_id){
+        $price = Book::where('id',$book_id)->value('price');
+        $sub = ($quantity + 1) * $price ;
+        $borrowedBook_update = BorrowedBook::where('id',$borrowed_id)->update(['quantity' => $quantity + 1,'sub_total' => $sub]);
+        return $borrowedBook_update;
+    }
+
+    public function qtyMinus($borrowed_id, $quantity,$book_id){
+        $price = Book::where('id',$book_id)->value('price');
+        $sub = ($quantity - 1) * $price;
+        $borrowedBook_update = BorrowedBook::where('id',$borrowed_id)->update(['quantity' => $quantity - 1,'sub_total' => $sub]);
+        
+        if($this->checkQuantity($borrowed_id) ==  '0'){
+            $this->destroy($borrowed_id);
+        }
+        return $borrowedBook_update;
+    }
+
+    public function rentalBook(){
+
+        $check_borrow = BorrowedBook::where('user_id','=',Auth()->id())->get();
+        $date_time= mktime(0, 0, 0, date("m"), date("d")+10, date("Y"));
+        $datetime_return =date("Y/m/d H:i:s", $date_time);
+
+        for($i= 0; $i<count($check_borrow) ; $i++){
+            BorrowedBook::whereIn('id', [$check_borrow[$i]->id])->update(['status' => 2,'datetime_borrow'=> Carbon::now(),'datetime_return' => $datetime_return]);
+        }
+    }
+
+    public function destroy($borrow_id){
+        
+        $borrow_destroy = BorrowedBook::destroy($borrow_id);
+
+        if($borrow_destroy){
+            return redirect(route('book_borrow.index'))->with('delete_success',__('message.delete_success'));
+        }else{
+            return back()->with('fail',__('message.fail'));
+        }
+    }
+
+    public function destroyAll(){
+        $check_borrow = BorrowedBook::where([['user_id','=',Auth()->id()],['status','=',0]])->get();
+        for($i= 0; $i<count($check_borrow) ; $i++){
+           $delete_all_borrow_book = BorrowedBook::whereIn('id', [$check_borrow[$i]->id])->get();
+           $borrow_destroy = BorrowedBook::destroy($delete_all_borrow_book);
+        }
+        if($borrow_destroy){
+            return redirect(route('book_borrow.index'))->with('delete_success',__('message.delete_success'));
+        }else{
+            return back()->with('fail',__('message.fail'));
+        }
```

Comment:
Nếu check lỗi từng cái thì phải đưa vào trong for mới đúng nhé, nếu đặt if ở đây nó chỉ check cái cuối cùng thôi